ansible-ipi-install/roles/node-prep/tasks/main.yml

```diff
@@ -1,6 +1,13 @@
 ---
 - include_tasks: 10_validation.yml
   tags: validation
+  
+- include_tasks: 15_validation_disconnected_registry.yml
+  delegate_to: localhost
+  when:
+    - "'registry_host' in groups"
+    - "groups['registry_host']"
```

Comment:
you can just check the length of the group when: groups['registry_host'] | length == 0

Reply:
we probably want to check if the group is defined like we have now

```
137 - fail: msg="The {{ item }} group is not defined. Please add [{{ item }}] to the inventory/hosts file"
138   when: "'{{ item }}' not in groups"
139   loop:
140     - masters
141     - workers
142   tags: validation
```
Need to add - registry_host . Sucky part about this is it will break everyone whom doesn't have this in their existing inventory file so not sure if thats the best way to proceed atm.

Reply:
Let me know if I'm misunderstanding your comment but the check currently is on its own and has conditionals on the existence of disconnected vars which I think probably is the way to go. If you don't have any disconnected vars defined then it seems like at that point you are pretty much straight up telling ansible that you actually do want the cluster to be connected (otherwise you would've mentioned disconnected in your config somewhere). 

The inability to omit it is probably due to the act of putting the disconnected checks in the same task as the general purpose checks and thus you can't specifically target them with a `when` to tell them to only run if they're trying to install disconnected.

Reply:
so i can add the registry_host group to the 10_validation.yml file and modify the main.ylm as you mention. However, that will require the registry_host group to be defined even if not using a disconnected registry.
The way I have it in the main.yml allows the playbook to run if the registry_host group isnot defined. My thoughts were this makes the inventory file less complicated if disconnected is not used.
Your thoughts?

Reply:
re-looking at this, the 
```
  when:
    - "'registry_host' in groups"
    - "groups['registry_host']"
```
as it stands is likely the way to go. I think when I commented it I overlooked that its checking if registry_host AND the group exist.

Reply:
> so i can add the registry_host group to the 10_validation.yml file and modify the main.ylm as you mention. However, that will require the registry_host group to be defined even if not using a disconnected registry.
> The way I have it in the main.yml allows the playbook to run if the registry_host group isnot defined. My thoughts were this makes the inventory file less complicated if disconnected is not used.
> Your thoughts?

your way is better John :) Keep it as you have it.